Makefile

```diff
@@ -126,34 +132,70 @@ endef
 print-go-version:
 	@go version
 
-update-goimports:
-	@printf $(COLOR) "Install/update goimports..."
-	@go install golang.org/x/tools/cmd/goimports@latest
-
-update-linters:
-	@printf $(COLOR) "Install/update linters..."
-	# When updating the version, update the golangci-lint GHA workflow as well.
-	@go install github.com/golangci/golangci-lint/cmd/golangci-lint@v1.57.2
-
-update-mockgen:
-	@printf $(COLOR) "Install/update mockgen tool..."
-	@go install github.com/golang/mock/mockgen@v1.7.0-rc.1
-
-update-gotestsum:
-	@printf $(COLOR) "Install/update gotestsum..."
-	@go install gotest.tools/gotestsum@v1.11
-
-update-proto-plugins:
-	@printf $(COLOR) "Install/update proto plugins..."
-	@go install google.golang.org/protobuf/cmd/protoc-gen-go@latest
-	@go install google.golang.org/grpc/cmd/protoc-gen-go-grpc@latest
-	@go install -modfile build/go.mod go.temporal.io/api/cmd/protoc-gen-go-helpers
-	@go install -modfile build/go.mod go.temporal.io/api/cmd/protogen
-
-update-proto-linters:
-	@printf $(COLOR) "Install/update proto linters..."
-	@go install github.com/googleapis/api-linter/cmd/api-linter@v1.32.3
-	@go install github.com/bufbuild/buf/cmd/buf@v1.6.0
+$(STAMPDIR):
+	@mkdir -p $(STAMPDIR)
+
+$(LOCALBIN):
+	@mkdir -p $(LOCALBIN)
+
+# When updating the version, update the golangci-lint GHA workflow as well.
+.PHONY: golangci-lint
+GOLANGCI_LINT_VERSION := v1.57.2
+GOLANGCI_LINT := $(LOCALBIN)/golangci-lint-$(GOLANGCI_LINT_VERSION)
+$(GOLANGCI_LINT): $(LOCALBIN)
+	$(call go-install-tool,$(GOLANGCI_LINT),github.com/golangci/golangci-lint/cmd/golangci-lint,$(GOLANGCI_LINT_VERSION))
+
+GOIMPORTS_VER := v0.20.0
+GOIMPORTS := $(LOCALBIN)/goimports-$(GOIMPORTS_VER)
```

Comment:
shouldn't we be using `build/go.mod` to track these versions? are we deprecating the `build` dir entirely? if so, remove it in this pr?

Reply:
I had trouble getting build/go.mod to do what I needed. I'd be happy to deprecate the build dir here

Reply:
yes, let's have a single mechanism at a time